File: crates/flowrt-codegen/tests/golden/sync_fusion_cpp/expected/app/stubs/cpp/odom_src.cpp
// FlowRT 管理参考模板（C++）。可删除重建；复制到用户 app/ 后再修改。

#include "flowrt_app/runtime_shell.hpp"

#include <memory>

namespace {

class OdomSrc final : public flowrt_app::OdomSrcInterface {
public:
    flowrt::Status on_tick(
        flowrt::Output<flowrt_app::Odom>& odom) override {
        odom.write(flowrt_app::Odom{});
        return flowrt::ok();
    }
};

}  // namespace

namespace flowrt_user {

flowrt_app::App build_app() {
    return flowrt_app::App(std::make_unique<OdomSrc>());
}

}  // namespace flowrt_user

File: crates/flowrt-codegen/tests/golden/sync_fusion_rust/expected/app/stubs/rust/odom_src.rs
// FlowRT 管理参考模板（Rust）。可删除重建；复制到用户 app/ 后再修改。

#[derive(Default)]
pub struct OdomSrc;

impl flowrt_app::components::OdomSrc for OdomSrc {
    fn on_tick(
        &mut self,
        odom: &mut flowrt::Output<flowrt_app::messages::Odom>,
    ) -> flowrt::Status {
        odom.write(flowrt_app::messages::Odom::default());
        flowrt::Status::Ok
    }
}

pub fn build_app() -> flowrt_app::runtime_shell::App {
    flowrt_app::runtime_shell::App::new(Box::new(OdomSrc::default()))
}
